src/main/java/com/dtolabs/rundeck/plugin/overthere/OTWinRMFileCopier.java

```diff
@@ -0,0 +1,805 @@
+                                        "Destination folder for remote scripts", false, null))
+            .property(
+                      PropertyBuilder.builder()
+                                     .string(CONFIG_PASSWORD_STORAGE_PATH)
+                                     .title("Password Storage")
+                                     .description(
+                                             "Key Storage Path for winrm Password.\n\n" +
+                                             "The path can contain property references like `${node.name}`."
+                                     )
+                                     .renderingOption(
+                                             StringRenderingConstants.SELECTION_ACCESSOR_KEY,
+                                             StringRenderingConstants.SelectionAccessor.STORAGE_PATH
+                                     )
+                                     .renderingOption(
+                                             StringRenderingConstants.STORAGE_PATH_ROOT_KEY,
+                                             "keys"
+                                     )
+                                     .renderingOption(
+                                             StringRenderingConstants.STORAGE_FILE_META_FILTER_KEY,
+                                             "Rundeck-data-type=password"
+                                     )
+                                     .build()
+            )
+            .mapping(CONFIG_AUTHENTICATION, PROJ_PROP_PREFIX + WINRM_AUTH_TYPE)
+            .mapping(CONFIG_PROTOCOL, PROJ_PROP_PREFIX + WINRM_PROTOCOL)
+            .mapping(CONFIG_CERT_TRUST, PROJ_PROP_PREFIX + WINRM_CERT_TRUST)
+            .mapping(CONFIG_HOSTNAME_VERIFY, PROJ_PROP_PREFIX + WINRM_HOSTNAME_TRUST)
+            .mapping(CONFIG_SPN_ADD_PORT, PROJ_PROP_PREFIX + WINRM_SPN_ADD_PORT)
+            .mapping(CONFIG_SPN_USE_HTTP, PROJ_PROP_PREFIX + WINRM_SPN_USE_HTTP)
+            .mapping(CONFIG_LOCALE, PROJ_PROP_PREFIX + WINRM_LOCALE)
+            .mapping(CONFIG_WINRM_TIMEOUT, PROJ_PROP_PREFIX + WINRM_TIMEOUT)
+            .mapping(CONFIG_TIMEOUT, PROJ_PROP_PREFIX + WINRM_CONNECTION_TIMEOUT_PROPERTY)
+            .mapping(CONFIG_PASSWORD_STORAGE_PATH, PROJ_PROP_PREFIX + WINRM_PASSWORD_STORAGE_PATH)
+            .mapping(CONFIG_DST_FILE, PROJ_PROP_PREFIX + FILE_COPY_DESTINATION_DIR)
+            .build();
+
+    public Description getDescription() {
+        return DESC;
+    }
+
+    @Override
+    public String copyFileStream(ExecutionContext context, InputStream inputStream, INodeEntry node, String destination) throws FileCopierException {
+        ConnectionOptions options = null;
+        String logprompt = "[" + SERVICE_PROVIDER_TYPE + ":" + node.extractHostname() + "] ";
+
+        if (null == context.getExecutionListener()) {
+            System.out.println(logprompt + " Bad plugin context!  NULL ExecutionListener");
```

Comment:
.log() instead of println()

Reply:
Can not here as the log() is precisely in the context.getExecutionListener()

Reply:
yuck. okay.